src/Storages/Statistics/StatisticsCountMinSketch.cpp

```diff
@@ -0,0 +1,88 @@
+#include <DataTypes/DataTypeLowCardinality.h>
+#include <DataTypes/DataTypeNullable.h>
+#include <IO/ReadHelpers.h>
+#include <IO/WriteHelpers.h>
+#include <Storages/Statistics/StatisticsCountMinSketch.h>
+
+#if USE_DATASKETCHES
+
+namespace DB
+{
+
+namespace ErrorCodes
+{
+extern const int ILLEGAL_STATISTICS;
+}
+
+static constexpr auto num_hashes = 8uz;
+static constexpr auto num_buckets = 2048uz;
+
+StatisticsCountMinSketch::StatisticsCountMinSketch(const SingleStatisticsDescription & stat_, DataTypePtr data_type_)
+    : IStatistics(stat_)
+    , sketch(num_hashes, num_buckets)
+    , data_type(data_type_)
+{
+}
+
+Float64 StatisticsCountMinSketch::estimateEqual(const Field & value) const
+{
+    if (auto float_val = IStatistics::getFloat64(value))
+        return sketch.get_estimate(&float_val.value(), 8);
+    if (auto string_val = IStatistics::getString(value))
+        return sketch.get_estimate(string_val->data(), string_val->size());
+    UNREACHABLE();
```

Comment:
That will unconditionally crash the server, let's instead throw a `LOGICAL_ERROR`.